src/Forms/FormTextarea.js

```diff
@@ -1,16 +1,58 @@
 import classnames from 'classnames';
 import { FORM_STATES } from '../utils/constants';
 import PropTypes from 'prop-types';
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 
-const FormTextarea = React.forwardRef(({ children, className, compact, disabled, disableStyles, readOnly, state, ...props }, ref) => {
+const FormTextarea = React.forwardRef(({
+    children,
+    className,
+    compact,
+    counterProps,
+    defaultValue,
+    disabled,
+    disableStyles,
+    maxLength,
+    onChange,
+    readOnly,
+    state,
+    value,
+    ...props }, ref) => {
 
     useEffect(() => {
         if (!disableStyles) {
             require('fundamental-styles/dist/textarea.css');
         }
     }, []);
 
+    const getInitialCharCount = () => {
+        if (typeof value === 'boolean' || value) {
+            return Math.min(value.toString().length, maxLength);
+        }
+        if (typeof children === 'string') {
+            return Math.min(children.length, maxLength);
+        }
+        if (defaultValue) {
+            return Math.min(defaultValue.toString().length, maxLength);
+        }
+        return 0;
+    };
+
+    const hasMaxLength = maxLength >= 0;
+
+    const [charCount, setCharCount] = useState(hasMaxLength ? getInitialCharCount() : 0);
+
+    const getMaxLengthText = () => {
+        let charsLeft = Math.max(maxLength - charCount, 0);
+        return `${charsLeft} character${charsLeft !== 1 ? 's' : ''} left`;
```

Comment:
I don't think our current way of handling allows for strings with arguments to conditionally render plural, but to Jenna's earlier point of using localizedText, this could be
 
```js
    const getMaxLengthText = () => {
        const {localizedText} = this.props;
        const {charactersLeftPlural, charactersLeftSingular} = localizedText;

        let charsLeft = Math.max(maxLength - charCount, 0);
	    if (charsLeft !== 1) {
            return `${charsLeft} ${charactersLeftPlural}`;
        }
        return `${charsLeft} ${charactersLeftSingular}`;
    }

// ...

TextArea.propTypes = {
   // ...
   localizedText: CustomPropTypes.i18n({
      charactersLeftPlural: PropTypes.string,
      charactersLeftSingular: PropTypes.string,
   })
}

TextArea.defaultProps = {
   // ...
   localizedText: {
      charactersLeftPlural: 'characters left',
      charactersLeftSingular: 'character left',
   }
}
```